modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/CommunicationTcpUtils.java

```diff
@@ -201,4 +205,161 @@ public static void failNode(ClusterNode nodeToFail,
             ", err=" + err +
             ", connectErrs=" + X.getSuppressedList(err) + ']');
     }
+
+    /**
+     * Base class for network interface matchers.
+     */
+    public abstract static class NetworkInterfaceMatcher {
+        /** Pattern of network interface. */
+        private final String iface;
+
+        /**
+         * Creates a new instance of network interface matcher.
+         * @param iface Pattern of network interface.
+         */
+        public NetworkInterfaceMatcher(String iface) {
+            this.iface = iface;
+        }
+
+        /**
+         * Gets the pattern of network interface.
+         * @return Pattern of network interface.
+         */
+        public String networkInterface() {
+            return iface;
+        }
+
+        /**
+         * Matches the given address against the pattern.
+         *
+         * @param addr Address to match.
+         * @return {@code true} if the address matches the pattern, {@code false} otherwise.
+         */
+        abstract boolean matches(InetAddress addr);
+
+        /** {@inheritDoc} */
+        @Override public String toString() {
+            return S.toString(NetworkInterfaceMatcher.class, this);
+        }
+    }
+
+    /**
+     * Matcher for IPv4 addresses.
+     */
+    public static class IPv4Matcher extends NetworkInterfaceMatcher {
+        private static final Pattern DIGIT_PATTERN = Pattern.compile("\\d{1,3}");
```

Comment:
This pattern matches something that is not a decimal digit, but it can be understood as 256-cimal digit. Could it be renamed to `OCTET` to avoid confusion?

Reply:
Yep, good catch!